### PR TITLE
fix: FOUC

### PR DESCRIPTION
**Pull request by Aslemammad**
part of #344 

Since the style scripts are modules, they run after the HTML parse phase, which causes the styles to flash. 

What I do in this PR, is I get the styles and I inject them as normal style tags in the first load and then for HMR, later they'll be removed automatically.

### Why not script tags?
The module type of the script tags and them importing things is limiting and won't let me to use blocking script tags.
blocking:
```html
<script />
```
non-blocking:
```html
<script type="module" />
```

And since the client-transformed CSS (that supports HMR) in vite is a JS module, we cannot use normal script tags. What can be done is server-transforming CSS, so we get the transformed plain CSS and then inject it in the style tags.

I also remove the spinner.

### CODE REVIEW

File: packages/waku/src/lib/plugins/vite-plugin-rsc-hmr.ts [hunk 1]
```diff
@@ -13,80 +19,44 @@ import.meta.hot = __vite__createHotContext(import.meta.url);
...
     script = document.createElement('script');
     script.type = 'module';
     script.text = code;
     script.setAttribute('waku-module-id', data.id);
     document.head.appendChild(script);
-    if (data.css) removeSpinner();
+    // avoid HMR flash by first applying the new and removing the old styles 
+    if (style) {
+      queueMicrotask(style.remove);
+    }
```
Comment (ojj1123) [inline packages/waku/src/lib/plugins/vite-plugin-rsc-hmr.ts:36]: I was wondering why you queued `style.remove` function?

Comment (Aslemammad) [reply packages/waku/src/lib/plugins/vite-plugin-rsc-hmr.ts:36]: So we do not have a FOUC in HMR, if we don't do this, styles will be removed, for a small bit of time, there are no styles until the new script tag applies the new styles. By queuing, we first apply the new script styles and then after that we remove the old styles.

**Review by dai-shi — APPROVED** [review]
The code looks good.

`npm run examples:dev:01_template` works well.
Do you notice `npm run website:dev` still has FOUC at first?

**Comment by Aslemammad** [comment]
Thank you

> Do you notice npm run website:dev still has FOUC at first?

I don't on my system! Is it still there?

**Comment by dai-shi** [comment]
@Aslemammad Yeah, as I tried, the `<style>` tag isn't present in the HTML.

Can anyone else try?

**Comment by Aslemammad** [comment]
> @Aslemammad Yeah, as I tried, the `<style>` tag isn't present in the HTML.
> 
> Can anyone else try?

Have u tried linking waku or updating it in the website directory? since the version is fixed there! Let me know

**Comment by dai-shi** [comment]
Ah, that's it. I forgot.

**Pull request merged**